isis/src/control/objs/BundleUtilities/BundleObservation.cpp

```diff
@@ -783,6 +783,433 @@ namespace Isis {
+                correctionUnitList.append("m/s");
+                break;
+              case 2:
+                parameterNamesList.append( str.arg("     ").arg("km/s^2") );
+                correctionUnitList.append("m/s^2");
+                break;
+              case 3:
+                parameterNamesList.append( str.arg("     ").arg("km/s^3") );
+                correctionUnitList.append("m/s^3");
+                break;
+              default:
+                  parameterNamesList.append(str.arg("     ").arg("km/s^"+toString(i) ) );
+                  correctionUnitList.append("m/s^"+toString(i));
+              }
+
+            }
+
+          }
+        }
+        if (nPointingCoefficients > 0) {
+          for (int i = 0; i < nPointingCoefficients; i++) {
+            finalParameterValues.push_back(coefRA[i] * RAD2DEG);
+            if (i == 0) {
+              parameterNamesList.append( str.arg(" RA  ").arg("dd") );
+              correctionUnitList.append("dd");
+            }
+            else {
+              switch(i) {
+
+              case 1:
+                parameterNamesList.append( str.arg("     ").arg("dd/s") );
+                correctionUnitList.append("dd");
+                break;
+              case 2:
+                parameterNamesList.append( str.arg("     ").arg("dd/s^2") );
+                correctionUnitList.append("dd/s^2");
+                break;
+              case 3:
+                parameterNamesList.append( str.arg("     ").arg("dd/s^3") );
+                correctionUnitList.append("dd/s^3");
+                break;
+              default:
+                  parameterNamesList.append(str.arg("     ").arg("dd/s^"+toString(i) ) );
+                  correctionUnitList.append("dd/s^"+toString(i));
+
+              }
+            }
+          }
+          for (int i = 0; i < nPointingCoefficients; i++) {
+            finalParameterValues.push_back(coefDEC[i] * RAD2DEG);
+            if (i == 0) {
+              parameterNamesList.append( str.arg("DEC  ").arg("dd") );
+              correctionUnitList.append("dd");
+
+            }
+            else {
+
+              switch(i) {
+
+              case 1:
+                parameterNamesList.append( str.arg("     ").arg("dd/s") );
+                correctionUnitList.append("dd/s");
+                break;
+              case 2:
+                parameterNamesList.append( str.arg("     ").arg("dd/s^2") );
+                correctionUnitList.append("dd/s^2");
+                break;
+              case 3:
+                parameterNamesList.append( str.arg("     ").arg("dd/s^3") );
+                correctionUnitList.append("dd/s^3");
+                break;
+              default:
+                  parameterNamesList.append(str.arg("     ").arg("dd/s^"+toString(i) ) );
+                  correctionUnitList.append("dd/s^"+toString(i) );
+
+              }
+            }
+          }
+          for (int i = 0; i < nPointingCoefficients; i++) {
```

Comment:
Same as above. It looks like the logic for the loops can be squashed into one loop? Or does this affect your ordering of parameterNamesList and correctionUnitList?

Reply:
It would affect the ordering, but I will look into it because anything that can simplify the function would be awesome.  This function follows the logic pathways of formatBundleOutputString so that could be simplified as well.

Reply:
Same as above: create lists (parameterNamesList and correctionUnitList) for each of the pointing arguments (RA, DEC and TWI) and combine them after the loop. In fact, after THIS loop may make sense to combine all list from this loop and the nPositionCoefficients from above. You would just need to pay attention to what order you are recombining them to keep the order necessary for the logic further down.